### PR TITLE
Release notes for Podman v3.3.0-RC2

### PR DESCRIPTION
**Pull request by mheon**
Also includes backports

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,104 @@
...
+- If an invalid subcommand is provided, similar commands to try will now be suggested in the error message.
+
+### Changes
+- The `podman system reset` command now removes non-Podman (e.g. Buildah and CRI-O) containers as well.
+- The new port forwarding offered by `podman machine` requires [gvproxy](https://github.com/containers/gvisor-tap-vsock) in order to function.
+- Podman will now automatically create the default CNI network if it does not exist, for both root and rootless users. This will only be done once per user - if the network is subsequently removed, it will not be recreated.
+- The `--root` option to Podman will not automatically clear all default storage options when set. Storage options can be set manually using `--storage-opt` ([#10393](https://github.com/containers/podman/issues/10393)).
+- The `install.cni` makefile option has been removed. It is no longer required to distribute the default `87-podman.conflist` CNI configuration file, as Podman will not automatically create it.
```
Comment (edsantiago) [inline RELEASE_NOTES.md:39]: "not automatically create it" -- should that be **now**? (I may be way off; my feedback is based on line 37 above).

**Comment by mheon** [comment]
Comments addressed

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,5 +1,104 @@
...
+- The `podman ps --filter` option's `network` filter now accepts a new value: `container:`, which matches containers that share a network namespace with a specific container ([#10361](https://github.com/containers/podman/issues/10361)).
+- The `podman diff` command can now accept two arguments, allowing two images or two containers to be specified; the diff between the two will be printed ([#10649](https://github.com/containers/podman/issues/10649)).
+- Podman can now optionally copy-up content from containers into volumes mounted into those containers earlier (at creation time, instead of at runtime) via the `prepare_on_create` option in `containers.conf` ([#10262](https://github.com/containers/podman/issues/10262)).
+- A new option, `--gpus`, has been added to `podman create` and `podman run` as a no-op for better compatibility with Docker. If the nvidia-container-runtime package is installed, GPUs should be automatically added to containers without using the flag.
+- If an invalid subcommand is provided, similar commands to try will now be suggested in the error message.
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply RELEASE_NOTES.md:32]: @ashley-cui PTAL

[the remark this answers is not in the document]
Comment (ashley-cui) [reply RELEASE_NOTES.md:32]: `podman build --secret` does work

[the remark this answers is not in the document]
Comment (mheon) [reply RELEASE_NOTES.md:32]: @rhatdan If you want Buildah features, you're going to need someone else to get those release notes written up and passed over to me. Takes days to just get Podman done, I can't handle all our dependencies as well.

[the remark this answers is not in the document]
Comment (mheon) [reply RELEASE_NOTES.md:33]: Added `build --secret`

**Comment by mheon** [comment]
@containers/podman-maintainers PTAL. Restarted three flakes, otherwise should be good.

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -1,5 +1,105 @@
...
+- The `podman pull` command can now pull multiple images at once (e.g. `podman pull fedora:34 ubi8:latest` will pull both specified images).
+- THe `podman cp` command can now copy files from one container into another directly (e.g. `podman cp containera:/etc/hosts containerb:/etc/`) ([#7370](https://github.com/containers/podman/issues/7370)).
+- The `podman cp` command now supports a new option, `--archive`, which controls whether copied files will be chown'd to the UID and GID of the user of the destination container.
+- The `podman stats` command now provides two additional metrics: Average CPU, and CPU time.
+- The `podman pod create` command supports a new flag, `--pid`, to specify the PID namespace of the pod. If specified, containers that join the pod will automatically share its PID namespace.
+- The `podman pod create` command supports a new flag, `--infra-name`, which allows the name of the pod's infra container to be set ([#10794](https://github.com/containers/podman/issues/10794)).
+- The `podman auto-update` command has had its output reformatted - it is now much more clear what images were pulled and what containers were updated.
```
Comment (TomSweeneyRedHat) [inline RELEASE_NOTES.md:20]:
```suggestion
- The `podman auto-update` command has had its output reformatted - it is now much clearer what images were pulled and what containers were updated.
```

File: RELEASE_NOTES.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply RELEASE_NOTES.md:33]: @mheon I've not thought about doing `buildah build` RA things in here, but it makes sense.  Going forward I'll try to get them to you, if not whack me upside the head.

[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply RELEASE_NOTES.md:33]: I didn't think this was Buildah specific, but Valenting added:

A dangling image is now considered dangling if it is “untagged”  and does not have children.  This now matches the definition used by Docker.

Did that get added to Podman too?  I'm not finding a note about dangling here @mheon

[the remark this answers is not in the document]
Comment (mheon) [reply RELEASE_NOTES.md:33]: It did, but I did not explicitly note the change in the definition of "dangling" in the release notes, and instead noted the bugs that were fixed by this (`podman image prune` being too aggressive). I don't think that the definition of dangling represents much of a user-facing change.

[the remark this answers is not in the document]
Comment (TomSweeneyRedHat) [reply RELEASE_NOTES.md:33]: fair enough, thanks for the follow up.

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

**Comment by mheon** [comment]
Now also includes a revert of pod `--cpus` flag, which will not be ready in time for 3.3

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**